core/test/matrix/permutation.cpp

```diff
@@ -155,4 +155,15 @@ TYPED_TEST(Permutation, CanBeCleared)
 }
 
 
+TYPED_TEST(Permutation, RecognizesDuplicatePermutationIndices)
+{
+    using index_type = typename TestFixture::index_type;
+    index_type data[] = {1, 0, 2, 2};
+
+    auto m = gko::matrix::Permutation<index_type>::create_const(
+        this->exec, gko::array<index_type>::const_view(this->exec, 4, data));
+
+    ASSERT_THROW(m->validate_data(), gko::InvalidData);
+}
+
 }  // namespace
```

Comment:
```suggestion
}


}  // namespace
```
same here. ref: https://github.com/ginkgo-project/ginkgo/wiki/Contributing-guidelines#whitespace